recipes/kyotocabinet/meta.yaml

```diff
@@ -0,0 +1,65 @@
+{% set name = "kyotocabinet" %}
+{% set version = "1.2.76" %}
+{% set sha256 = "812a2d3f29c351db4c6f1ff29d94d7135f9e601d7cc1872ec1d7eed381d0d23c" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: http://fallabs.com/kyotocabinet/pkg/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - kcdbext.patch
+    - check_poly.patch
```

Comment:
Please add some comments above these patches explaining why their needed. Also have you raised these upstream?

Reply:
Hi @jakirkham,

First of all, I have to say, that _kyotocabinet_ has noot been maintained by its creators (FAL Labs) since 2012. I have recently discovered a fork by Cloudfare Inc. here on Github, which did continue maintaining it for some time, but the last commit there is 2 years old, and there is also a pull request pending for more than a year now (in fact, for a patch similar to my _kcdbext.patch_), so not much activity there either.

My colleagues and I have now created a new fork (https://github.com/trustyou/kyotocabinet) and already checked in the changes from _kcdbext.patch_ there, but as of yet, we are unsure if we are actually going to maintain this fork.

Now to the patches themselves.

1. The already mentioned _kcdbext.patch_ is needed for compatibility with newer GCC versions (GCC >= 4, unless I am mistaken). Without this patch, the code cannot be compiled.

2. The _check_poly.patch_ is of a different sort, in that it has nothing to do with the code itself. For some reason, the _check_poly_ task of the Makefile kept failing on *circleci*, although it did work for me locally. As this is rather difficult to debug, especially, since I cannot reproduce the problem, I decided to just deactivate this one Makefile task, by commenting it in the _Makefile.in_ file, which is essentially what this patch does.

I will add the comments about the patches to the _meta.yaml_ file, as well.

Reply:
Ah ok. Thanks for the clarifications. Agree that upstreaming the patches is not a particularly useful exercise at this stage. Thanks for adding the comments as well.